### PR TITLE
Smooth moving enhancement

### PR DESCRIPTION
**Pull request by hiba9201**
Made smooth moving via updating translate every 50ms
Later can try to do it with request animation frame and animate only visible vehicles

About visible vehicles I'm not sure, because as far as I remember leaflet should draw only visible elements, and we use leaflet's Markers, so everything should be okay

Next steps:
- animate vehicles on coordinates' update so they would not jump but quickly move to actual coordinate
- if new coordinates are behind drawn vehicle, so slow down the vehicle
- request animation frame
- fix bugs with update on zoom

### CODE REVIEW

File: components/Map/Vehicles/Item/MapVehiclesItem.utils.ts [hunk 1]
```diff
@@ -1,15 +1,63 @@
...
-    const velocityPixelsPerSecond = velocityMetersPerSecond / scale;
+const intervalById: {
+    [id: number]: number;
+} = {};
 
-    return seconds * velocityPixelsPerSecond;
-};
+function round(num: number) {
+    return Math.round((num + Number.EPSILON) * 10000) / 10000;
+}
```
Comment (hiba9201) [inline components/Map/Vehicles/Item/MapVehiclesItem.utils.ts:9]: Thinking about adding parameter with count of decimal places

File: components/Map/Vehicles/Item/MapVehiclesItem.utils.ts [hunk 2]
```diff
@@ -1,15 +1,63 @@
...
+    const distance = (velocity * ANIMATION_INTERVAL) / 1000;
     const angleInRad = (course * Math.PI) / 180;
 
-    return [Math.cos(angleInRad) * distance, Math.sin(angleInRad) * distance];
-};
+    return [round(Math.cos(angleInRad) * distance), round(Math.sin(angleInRad) * distance)];
+}
+
+function getVelocityInPixelsPerSecond(velocity: number, scale: number) {
+    const velocityMetersPerSecond = (velocity * 1000) / 3600;
```
Comment (hiba9201) [inline components/Map/Vehicles/Item/MapVehiclesItem.utils.ts:19]: Magical numbers to constants (1000 – meters in 1 kilometer, 3600 seconds in 1 hour)

File: components/Map/Vehicles/Item/MapVehiclesItem.utils.ts [hunk 3]
```diff
@@ -1,15 +1,63 @@
...
+
+    vehicle.style.transform = `translate3d(${x}px, ${y}px, 0px)`;
+}
+
+export function startMoveInDirection(
+    direction: number,
+    vehicle: HTMLElement,
+    velocity: number,
+    scale: number,
+) {
```
Comment (hiba9201) [inline components/Map/Vehicles/Item/MapVehiclesItem.utils.ts:46]: Object parameter

File: components/Map/Vehicles/Item/MapVehiclesItem.utils.ts [hunk 4]
```diff
@@ -1,15 +1,63 @@
...
+    return [round(Math.cos(angleInRad) * distance), round(Math.sin(angleInRad) * distance)];
+}
+
+function getVelocityInPixelsPerSecond(velocity: number, scale: number) {
+    const velocityMetersPerSecond = (velocity * 1000) / 3600;
+
+    return velocityMetersPerSecond / scale;
+}
+
+function moveInDirection(direction: number, vehicle: HTMLElement, velocity: number, scale: number) {
```
Comment (hiba9201) [inline components/Map/Vehicles/Item/MapVehiclesItem.utils.ts:24]: Object parameter

File: components/Map/Vehicles/Item/MapVehiclesItem.utils.ts [hunk 5]
```diff
@@ -1,15 +1,63 @@
...
 
-export const getDeltaCoords = (velocity: number, scale: number, course: number) => {
-    const distance = getDistanceInPixels(9, velocity, scale);
+function getDeltaCoords(velocity: number, course: number) {
+    const distance = (velocity * ANIMATION_INTERVAL) / 1000;
     const angleInRad = (course * Math.PI) / 180;
 
-    return [Math.cos(angleInRad) * distance, Math.sin(angleInRad) * distance];
-};
+    return [round(Math.cos(angleInRad) * distance), round(Math.sin(angleInRad) * distance)];
```
Comment (a-chabin) [inline components/Map/Vehicles/Item/MapVehiclesItem.utils.ts:15]: Let's add comment with explanation

**Pull request merged**